### PR TITLE
use nuget packages in samples instead of expecting visual studio to be installed

### PR DESCRIPTION
**Pull request by jmarolf**
## Summary

Describe your changes here.

Fixes #Issue_Number, dotnet/docs#Issue_Number or dotnet/dotnet-api-docs#Issue_Number (if available)

### CODE REVIEW

**Review by Youssef1313 — COMMENTED** [review]
Can you update that in [dotnet/roslyn-sdk](https://github.com/dotnet/roslyn-sdk/blob/main/src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/Diagnostic/Vsix/Deployment.csproj) (and the parallel VB)?

Also the sample in dotnet/docs should be kept in sync.

Adding @sharwell as well. He has been helping review changes for this particular sample.

**Comment by BillWagner** [comment]
@Youssef1313 

> Can you update that in dotnet/roslyn-sdk (and the parallel VB in roslyn-sdk)?

> Also the sample in dotnet/docs should be kept in sync.

I was going to open a new PR for those once we verified this is the solution. (Or you can, now that we see this works).

I'll merge this later today, unless @sharwell has comments on why we shouldn't.

**Review by sharwell — CHANGES_REQUESTED** [review]
I do not believe this change is correct (it wasn't for dev16, not yet sure about dev17).

File: csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Vsix/MakeConst.Vsix.csproj
Comment (sharwell) [inline csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Vsix/MakeConst.Vsix.csproj:22]: 📝 This package _defines_ `VSToolsPath` and places Microsoft.VsSDK.targets in the NuGet cache.

File: csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Vsix/MakeConst.Vsix.csproj
Comment (sharwell) [inline csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Vsix/MakeConst.Vsix.csproj:22]: 📝 This package is not necessary. Excluding runtime assets is also not necessary.

File: csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Vsix/MakeConst.Vsix.csproj
Comment (sharwell) [inline csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Vsix/MakeConst.Vsix.csproj:39]: 📝 This is not an automatic import. Failure to include this line breaking packaging and deployment.

**Review by sharwell — CHANGES_REQUESTED** [review]
Prior to this change, the build already used NuGet packages and did not expect the Visual Studio SDK to be installed. After this change, the packaging and deployment are broken. We should not merge this.

**Comment by sharwell** [comment]
I haven't seen the specific build error that led to this change, but it's my hypothesis that the change you're looking for here is just the addition of this property group to the VSIX project:

```xml
<PropertyGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
  <!-- This property disables extension deployment for command line builds; required for some CI -->
  <DeployExtension>False</DeployExtension>
</PropertyGroup>
```

**Comment by jmarolf** [comment]
updated PR here: https://github.com/dotnet/samples/pull/4594

**Pull request closed without merging**
